### ISSUE TITLE
Consolidate `publishedAt` and `config.publishedAt` fields

### ISSUE THREAD

**Issue by marcelgerber** [owid/owid-grapher#3345]
## Description

There are currently two fields on the `charts` table that (seem to) indicate whether a chart is published:
- `publishedAt`, which is the date when a chart was (first?) published
- `config.isPublished`, a boolean field that actually reflects whether a chart is published or not

It seems that `publishedAt` doesn't get reset when a chart gets unpublished.
This makes it so that there are currently 244 charts with a `publishedAt` date that are not actually public ([see Datasette](https://owid-datasette-y43qr.ondigitalocean.app/owid?sql=select%0D%0A++id%2C%0D%0A++slug%2C%0D%0A++config+-%3E%3E+%22%24.isPublished%22%2C%0D%0A++publishedAt%0D%0Afrom%0D%0A++charts%0D%0Awhere%0D%0A++%28%0D%0A++++publishedAt+is+not+null%0D%0A++++and+config+-%3E%3E+%22%24.isPublished%22+is+null%0D%0A++%29)).

This leads to issues like #3344, where we were using the wrong field to check for published status.

## We should

- either make it so `publishedAt` gets reset whenever a chart is unpublished,
- or, at the very least, check all uses of the `publishedAt` field, and validate that their use is actually justified and doesn't include "false-positive" charts

**Comment by danyx23** [comment]
Yes good call. We have a similar state of affairs with `posts` and `posts_gdocs` - though not quite as bad as the duplicated information boolean is just a separate column instead of nested in some json.

We should probably go with the `publishedAt === null` means unpublished solution in all of these cases.
